Reorder image card elements

maxgds opened the pull request:
## What

https://trello.com/c/mprRX2H8/376-confusing-order-of-content-for-featured-items-on-organisation-pages

Previously the source content for the image card was in a confusing order. Although it was clear from the design which elements belonged together, this was not reflected in the underlying page, with the heading coming after the image and contextual text. This switches the elements around and then uses CSS to realign them as before.

Order before:
- image
- context
- heading
- description and links

Order afterwards: 
- heading
- context
- description and links
- image

## Why
This improves the semantics of the page source and means that users who linearise the page will see the content and headings more clearly associated.

## Visual Changes
There are no intended visual changes. These screenshots are the updated code plugged into a collections page (http://collections.dev.gov.uk/government/organisations/department-for-business-energy-and-industrial-strategy) :

![Screenshot 2020-09-17 at 14 52 46](https://user-images.githubusercontent.com/31649453/93480773-b674b800-f8f5-11ea-884f-898666dcebd2.png)
![Screenshot 2020-09-17 at 14 53 01](https://user-images.githubusercontent.com/31649453/93480781-b7a5e500-f8f5-11ea-9b1a-f5167a10d718.png)
![Screenshot 2020-09-17 at 14 53 26](https://user-images.githubusercontent.com/31649453/93480782-b83e7b80-f8f5-11ea-86a0-d8f595a40ed7.png)

andysellick commented on CHANGELOG.md:
```diff
@@ -10,6 +10,7 @@
 
 ## Unreleased
 
+* Re-order elements in image card source ([PR #1694](https://github.com/alphagov/govuk_publishing_components/pull/1695)) FIX
```
Can this match the PR title?

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss:
```diff
@@ -3,6 +3,21 @@
+  @include govuk-media-query($from: mobile, $until: tablet) {
+    display: block;
+
+    .gem-c-image-card__text-wrapper {
```
Something's changed here in the layout on mobile, the gap between the image and the text has increased.

Before | After
------ | ------
<img width="428" alt="Screenshot 2020-09-21 at 08 28 55" src="https://user-images.githubusercontent.com/861310/93741557-92122780-fbe4-11ea-8a83-cf96660b7ec9.png"> | <img width="398" alt="Screenshot 2020-09-21 at 08 29 00" src="https://user-images.githubusercontent.com/861310/93741575-9b9b8f80-fbe4-11ea-8765-019d3f456547.png">
<img width="430" alt="Screenshot 2020-09-21 at 08 29 09" src="https://user-images.githubusercontent.com/861310/93741593-a35b3400-fbe4-11ea-859a-2a18eef935d1.png"> | <img width="390" alt="Screenshot 2020-09-21 at 08 29 17" src="https://user-images.githubusercontent.com/861310/93741618-ad7d3280-fbe4-11ea-8580-7a6874f72171.png">

andysellick commented on app/views/govuk_publishing_components/components/_image_card.html.erb:
```diff
@@ -41,10 +38,10 @@
         <p class="gem-c-image-card__metadata"><%= card_helper.metadata %></p>
       <% end %>
     </div>
+    <%= card_helper.image %>
```
I think this might need a bit more logic as it's broken the layout on mobile when there is no image.

Before | After
------ | ------
<img width="397" alt="Screenshot 2020-09-21 at 08 31 27" src="https://user-images.githubusercontent.com/861310/93741881-1795d780-fbe5-11ea-925c-a3e94329854a.png"> | <img width="385" alt="Screenshot 2020-09-21 at 08 31 32" src="https://user-images.githubusercontent.com/861310/93741899-1ebce580-fbe5-11ea-8fe4-457357a045e8.png">

andysellick reviewed (changes requested):
Unfortunately this change has slightly broken the layout in a couple of places.

maxgds commented:
@andysellick I've addressed your comments. To fix the no-image variant at mobile I've had to introduce an extra class, so I've added a test for its presence too. Would appreciate another look when you have time.

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss:
```diff
@@ -3,6 +3,22 @@
   @include govuk-text-colour;
   position: relative;
   margin-bottom: govuk-spacing(6);
+  display: flex;
```
IE10 and below doesn't seem to support the `flex` styles here properly, resulting in the title (and text) appearing before the image. Might be able to fix with some custom IE stuff.

<img width="972" alt="Screenshot 2020-09-21 at 15 27 44" src="https://user-images.githubusercontent.com/861310/93779775-3b760f00-fc1f-11ea-93e4-020d1fa061d9.png">

maxgds replied:
I've added -ms prefix versions and it now works in IE10 but not earlier. I think that's fine, the pages are perfectly legible and understandable in the old versions.

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss:
```diff
@@ -135,6 +151,10 @@
 
 .gem-c-image-card--large.gem-c-image-card {
+  display: flex;
+  @include govuk-media-query($from: tablet) {
```
Can we keep the `display` and `margin` selectors together above the `@include` bit?

maxgds replied:
Fixed

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/_image-card.scss:
```diff
@@ -180,3 +201,11 @@
+.gem-c-image-card--no-image {
+  .gem-c-image-card__text-wrapper {
+    @include govuk-media-query($from: mobile, $until: tablet) {
+      float: left;
```
This is better, but still not quite right, missing some spacing on the left.

<img width="393" alt="Screenshot 2020-09-21 at 15 28 18" src="https://user-images.githubusercontent.com/861310/93780074-9d367900-fc1f-11ea-9677-671987c4d2b2.png">

maxgds replied:
Fixed

maxgds commented:
@andysellick Could you take another look, please?
